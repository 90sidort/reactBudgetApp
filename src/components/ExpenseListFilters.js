import React from 'react'
import { connect } from 'react-redux'
import {
    setTextFilter,
    sortByDate,
    sortByAmount,
    setStartDate,
    setEndDate,
    setMinAmount,
    setMaxAmount
} from '../actions/filters'
import { DateRangePicker } from 'react-dates'
import ReactSlider from 'react-slider'

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    }
    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }))
    }
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }
    onSortChange = (e) => {
        if(e.target.value === 'amount'){this.props.sortByAmount()}
        else {this.props.sortByDate()}
    }
    OnValuesChange = (value) => {
        this.props.setMinAmount(value[0])
        this.props.setMaxAmount(value[1])
    }
    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input
                            className="text-input"
                            type="text"
                            value={this.props.filters.text}
                            onChange={this.onTextChange}
                            placeholder="Filter expenses"
                        />
                    </div>
                    <div className="input-group__item">
                        <select value={this.props.filters.sortBy} onChange={this.onSortChange} className="text-select">
                            <option value="date">Date</option>
                            <option value="amount">Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker
                            startDate={this.props.filters.startDate}
                            endDate={this.props.filters.endDate}
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.calendarFocused}
                            onFocusChange={this.onFocusChange}
                            numberOfMonths={1}
                            isOutsideRange={() => false}
                            showClearDates={true}
                        />
                    </div>
                </div>
                <div>
                    <ReactSlider
                        className="horizontal-slider"
                        thumbClassName="example-thumb"
                        trackClassName="example-track"
                        defaultValue={[this.props.filters.minAmount, this.props.filters.maxAmount]}
                        step={100}
                        max={10000}
                        minDistance={100}
                        ariaLabel={['Lower thumb', 'Upper thumb']}
                        ariaValuetext={state => `Thumb value ${state.valueNow}`}
                        renderThumb={(props, state) => <div {...props}>{state.valueNow}</div>}
                        pearling
                        minDistance={10}
                        onChange={this.OnValuesChange}
                    />
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({filters: state.filters})

const mapDispatchToProps = (dispatch) => ({
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setMinAmount: (minValue) => dispatch(setMinAmount(minValue)),
    setMaxAmount: (maxValue) => dispatch(setMaxAmount(maxValue)),
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)