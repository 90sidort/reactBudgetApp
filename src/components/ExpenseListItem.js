import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'

const ExpenseListItem = ({id, description, note, amount, createdAt}) => {
    return (
        <Link to={`/edit/${id}`} className="list_item">
            <div>
                <h3 className="list_item_description">{description}</h3>
                { note.length > 0 && <p className="list_item_notes"><small>{note}</small></p> }
                <span>{moment(createdAt).format('Do MMMM YYYY')}</span>
            </div>
            <h3 className="list_item_amount">{numeral(amount / 100).format('$0,0.00')}</h3>
        </Link> 
    )
}

export default ExpenseListItem