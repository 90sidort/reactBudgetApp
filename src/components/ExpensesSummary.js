import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/selectExpensesTotal'
import numeral from 'numeral'

export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
    const isSingle = expensesCount === 1 ? 'expense' : 'expenses'
    const formattedSum = numeral(expensesTotal / 100).format('$0,0.00')
    return(
        <div className="pageHeader">
            <div className="summaryContent">
                <p className="pageHeaderTitle">Viewing <span>{expensesCount}</span> {isSingle}, totaling <span>{formattedSum}</span>.</p>
            </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)

    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)