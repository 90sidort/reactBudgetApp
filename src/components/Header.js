import { NavLink } from 'react-router-dom'
import React from 'react';
import { connect } from 'react-redux'
import { startLogOut } from '../actions/auth'

export const Header = ({ startLogOut }) => (
    <header className="header">
    <div className="headerContent">
        <div className="headerMenu">
            <h1>Expensify</h1>
            <NavLink to="/dashboard" activeClassName="is-active" className="linkElement">Dashboard</NavLink>
            <NavLink to="/create" activeClassName="is-active" className="linkElement">Create expense</NavLink>
            <NavLink to="/help" activeClassName="is-active" className="linkElement">Help</NavLink>
        </div>
            <button className="buttonLogout" onClick={startLogOut}>Logout</button>
        </div>
    </header>
)

const mapDispatchToProps = (dispatch) => ({
    startLogOut: () => dispatch(startLogOut())
})

export default connect(undefined, mapDispatchToProps)(Header)