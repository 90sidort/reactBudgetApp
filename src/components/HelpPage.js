import React from 'react'

const HelpPage = () => (
    <div>
        This is my help Page
    </div>
)

export default HelpPage