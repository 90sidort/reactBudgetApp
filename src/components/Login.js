import React from 'react'
import { connect } from 'react-redux'
import { startLogin, startLogUser } from '../actions/auth'

export const Login = ({ startLogin, startLogUser }) => (
         <div className="boxLayout">
         <div className="boxLayoutBox">
            <h1 className="boxLayoutTitle">Expensify</h1>
            <p>Control your finance!</p>
            <button className="buttonLogin" onClick={startLogin}>Log in with Google</button>
            <button className="buttonLogin" onClick={startLogUser}>Test user</button>
         </div>
        </div>
)

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin()),
    startLogUser: () => dispatch(startLogUser())
})

export default connect(undefined, mapDispatchToProps)(Login)