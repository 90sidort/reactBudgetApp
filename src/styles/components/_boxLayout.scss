.boxLayout {
    background: url('/images/background.jpg');
    height: 100vh;
    width: 100vw;
    background-size: cover;
    display: flex;
    justify-content: center;
    align-items: center;
    opacity: 0.9;
}

.boxLayoutBox {
    background: fade-out($color: white, $amount: .15);
    border-radius: 5px;
    text-align: center;
    width: 25rem;
    padding: $l-size $m-size;
}

.boxLayoutTitle {
    margin: 0 0 $m-size 0;
    line-height: 1;
}

.buttonLogin {
    background: #1c88bf;
    border-radius: 3px;
    border-color: rgb(79, 79, 223);
    color: fade-out($color: white, $amount: .15);
    margin: 2px;
    padding: 3px;
    font-size: 2rem;
}