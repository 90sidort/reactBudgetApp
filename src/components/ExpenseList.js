import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
import selectExpenses from '../selectors/expenses'

export const ExpenseList = (props) => {

    return(
        <div className="content-container list_body">
        <div className="list-header">
            <div className="show_for_mobile">Expenses</div>
            <div className="show_for_desktop">Expense</div>
            <div className="show_for_desktop">Amount</div>
        </div>
            {
                props.expenses.length === 0 ? (
                    <p className="list_item_empty">No expenses</p>
                ) : (
                    props.expenses.map((expense) => (
                    <ExpenseListItem key={expense.id} {...expense}/>
                ))
                )
            }
        </div>
    )
}

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}

const ConnectedExpenseList = connect(mapStateToProps)(ExpenseList) 

export default ConnectedExpenseList