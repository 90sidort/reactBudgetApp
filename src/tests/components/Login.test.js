import React from 'react'
import { shallow } from 'enzyme'
import { Login } from '../../components/Login'

test('Should correctly render component', () => {
    const result = shallow(<Login startLogin={() => {}}/>)
    expect(result).toMatchSnapshot()
})

test('Should call startLogin on button click', () => {
    const startLoginSpy = jest.fn()
    const wrapper =  shallow(<Login startLogin={startLoginSpy}/>)
    wrapper.find('button').simulate('click')
    expect(startLoginSpy).toHaveBeenCalled()
})