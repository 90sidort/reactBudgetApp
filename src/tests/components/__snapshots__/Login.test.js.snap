// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render component 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Log in
  </button>
</div>
`;
